polyply/src/gen_coords.py

```diff
@@ -31,77 +31,76 @@
 from .annotate_ligands import AnnotateLigands, parse_residue_spec, _find_nodes
 from .build_file_parser import read_build_file
 
-def split_residues(molecule, split):
-    max_resid = len(molecule.nodes)
-    for split_string in split:
-        max_resid = molecule.split_residue(split_string, max_resid)
-
-def gen_coords(args):
-    # Read in the topology
-    topology = Topology.from_gmx_topfile(name=args.name, path=args.toppath)
-    topology.preprocess()
-
+def find_starting_node_from_spec(topology, start_nodes):
+    start_dict = {mol_idx:None for mol_idx, _ in enumerate(topology.molecules)}
+    for start in start_nodes:
+        res_spec = parse_residue_spec(start)
+        if 'mol_idx' in res_spec:
+            mol_idx = res_spec['mol_idx']
+            node = list(_find_nodes(topology.molecules[mol_idx], res_spec))[0]
+            start_dict[mol_idx] = node
+        else:
+            for idx, molecule in enumerate(topology.molecules):
+                if molecule.mol_name == res_spec['molname']:
+                    node = list(_find_nodes(molecule, res_spec))[0]
+                    start_dict[idx] = node
+    return start_dict
+
+def _check_molecules(molecules):
+    """
+    Helper method which raises an IOError
+    if any molecule in `molecules` is
+    disconnected.
+    """
     # check if molecules are all connected
-    for molecule in topology.molecules:
+    for molecule in molecules:
         if not nx.is_connected(molecule):
             msg = ('\n Molecule {} consistes of two disconnected parts. '
                    'Make sure all atoms/particles in a molecule are '
                    'connected by bonds, constraints or virual-sites')
             raise IOError(msg.format(molecule.name))
 
+def gen_coords(args):
+    # Read in the topology
+    print("reading topology")
+    topology = Topology.from_gmx_topfile(name=args.name, path=args.toppath)
+    print("processing it")
```

Comment:
leftover prints? If you want to provide user feedback/output, use logging

Reply:
yes but I feel implementing a logger at the moment would bloat this PR even more.

Reply:
Agreed. Leave them for hte next PR Put a #TODO next to it